### PR TITLE
[remote-shuffle]Refactor

### PR DESCRIPTION
**Pull request by gczsjdy**
## What changes were proposed in this pull request?

This PR 
- renames some classes, to be more clear, like `ShuffleRemoteSorter` -> `RemoteUnsafeShuffleSorter`
- Eliminates some comments containing 'HDFS', which should instead be general: for all Hadoop compatible filesystem, resolving https://github.com/Intel-bigdata/RemoteShuffle/issues/46
- Uses vanilla Spark's configuration as much as possible, avoiding to create new ones, like -> turn to using spark.reducer.maxBlocksInFlightPerAddress

## How was this patch tested?

Existed tests

### CODE REVIEW

**Comment by gczsjdy** [comment]
cc @carsonwang Could you please help me review this?

File: oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala [hunk 1]
```diff
@@ -40,7 +40,7 @@ object RemoteShuffleConf {
       .createWithDefault("/shuffle")
 
   val DFS_REPLICATION: ConfigEntry[Int] =
-    ConfigBuilder("spark.shuffle.remote.dfsReplication")
+    ConfigBuilder("spark.shuffle.remote.hdfs.replication")
```
Comment (carsonwang) [inline oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala:43]: How can we support this config in DAOS as well?

File: oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ object RemoteShuffleConf {
       .createWithDefault("hdfs://localhost:9001")
 
   val STORAGE_HDFS_MASTER_UI_PORT: ConfigEntry[String] =
-    ConfigBuilder("spark.shuffle.remote.storageMasterUIPort")
+    ConfigBuilder("spark.shuffle.remote.hdfs.storageMasterUIPort")
```
Comment (carsonwang) [inline oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala:31]: also update the doc?

File: oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala (hunk 1 above)
Comment (gczsjdy) [reply oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala:43]: It' easy to put a config entry inside Hadoop `Configuration`, then DAOS-Hadoop should parse it.

File: oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala (hunk 2 above)
Comment (gczsjdy) [reply oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala:31]: No problem.

Comment (gczsjdy) [reply oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala:31]: Done

File: oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala (hunk 1 above)
Comment (gczsjdy) [reply oap-shuffle/remote-shuffle/src/main/scala/org/apache/spark/shuffle/remote/RemoteShuffleConf.scala:43]: We probably don't need to do this at present, let's wait until DAOS-Hadoop support it.

**Pull request merged**